distribution/config/nginx.conf

```diff
@@ -1,60 +1,161 @@
-    server localhost:8081;
+# Includes files with directives to load dynamic modules.
+include /etc/nginx/modules/*.conf;
+
+events {
+    # The maximum number of simultaneous connections that can be opened by
+    # a worker process.
+    worker_connections 1024;
 }
 
-server {
-    listen 80 default_server;
-    listen [::]:80 default_server;
-    root /evergreen/www;
+http {
+    # Includes mapping of file name extensions to MIME types of responses
+    # and defines the default type.
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream;
 
-    error_log /dev/stdout info;
-    error_page 502 /502.html;
-    error_page 503 /503.html;
-    error_page 521 /521.html;
+    # Don't tell nginx version to clients.
+    server_tokens off;
 
-    location ^~ /socket.io {
-        proxy_pass http://evergreen;
-    }
+    # Specifies the maximum accepted body size of a client request, as
+    # indicated by the request header Content-Length. If the stated content
+    # length is greater than this size, then the client receives the HTTP
+    # error code 413. Set to 0 to disable.
+    client_max_body_size 1m;
 
-    location / {
-        if ( -f /evergreen/booting.txt ) {
-            # Used by Cloudflare as "Web Server is down"
-            #
-            # We're using in this case to convey to the end user that the
-            # Jenkins environment is booting
-            return 521;
-        }
-        proxy_set_header Host $host:$server_port;
-        proxy_redirect http://localhost/ http://localhost:8080/;
-        proxy_pass http://jenkins;
-    }
+    # Timeout for keep-alive connections. Server will close connections after
+    # this time.
+    keepalive_timeout 65;
 
-    location /evergreen {
-        rewrite /evergreen/(.*) /$1  break;
-        proxy_pass http://evergreen;
-    }
+    # Sendfile copies data between one FD and other from within the kernel,
+    # which is more efficient than read() + write().
+    sendfile on;
 
-    # Preventing route recursion
-    location = /404.html {
-        internal;
-    }
-    # Bad Gateway
-    location = /502.html {
-        internal;
+    # Don't buffer data-sends (disable Nagle algorithm).
+    # Good for sending frequent small bursts of data in real time.
+    tcp_nodelay on;
+
+    # Specifies that our cipher suits should be preferred over client ciphers.
+    ssl_prefer_server_ciphers on;
+
+    # Enables a shared SSL cache with size that can hold around 8000 sessions.
+    ssl_session_cache shared:SSL:2m;
+
+    # Enable gzipping of responses.
+    gzip on;
+
+    # Set the Vary HTTP header as defined in the RFC 2616.
+    gzip_vary on;
+
+    # Enable checking the existence of precompressed files.
+    #gzip_static on;
+
+    # Specifies the main log format.
+    log_format main '$remote_addr - $remote_user [$time_local] "$request" '
+            '$status $body_bytes_sent "$http_referer" '
+            '"$http_user_agent" "$http_x_forwarded_for"';
```

Comment:
I've stored in the back of my mind that at some point, it'd be good to try and align the logging formats (at least the timestamp) between the various processes running in the container.